lib/services/ios-project-service.ts

```diff
@@ -388,27 +413,75 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 			await this.setupSigningForDevice(projectRoot, buildConfig, projectData);
 		}
 
-		if (buildConfig && buildConfig.codeSignIdentity) {
-			args.push(`CODE_SIGN_IDENTITY=${buildConfig.codeSignIdentity}`);
-		}
+		// These are now set in the .pbxproj, avoid passing them as arguments!
+		// if (buildConfig && buildConfig.codeSignIdentity) {
+		// 	args.push(`CODE_SIGN_IDENTITY=${buildConfig.codeSignIdentity}`);
+		// }
 
-		if (buildConfig && buildConfig.mobileProvisionIdentifier) {
-			args.push(`PROVISIONING_PROFILE=${buildConfig.mobileProvisionIdentifier}`);
-		}
+		// if (buildConfig && buildConfig.mobileProvisionIdentifier) {
+		// 	args.push(`PROVISIONING_PROFILE=${buildConfig.mobileProvisionIdentifier}`);
+		// }
 
-		if (buildConfig && buildConfig.teamId) {
-			args.push(`DEVELOPMENT_TEAM=${buildConfig.teamId}`);
-		}
+		// if (buildConfig && buildConfig.teamId) {
+		// 	args.push(`DEVELOPMENT_TEAM=${buildConfig.teamId}`);
+		// }
+
+		await this.xcodebuild(args, projectRoot, buildConfig.buildOutputStdio);
+		await this.createIpa(projectRoot, projectData, buildConfig);
+	}
 
-		// this.$logger.out("xcodebuild...");
-		await this.$childProcess.spawnFromEvent("xcodebuild",
-			args,
+	private async xcodebuild(args: string[], cwd: string, stdio: any = "inherit"): Promise<ISpawnResult> {
+		const localArgs = [...args];
+		const xcodeBuildVersion = await this.getXcodeVersion();
+		try {
+			if (helpers.versionCompare(xcodeBuildVersion, "9.0") >= 0) {
+				localArgs.push("-allowProvisioningUpdates");
+			}
+		} catch (e) {
+			console.warn("Failed to use xcodebuild with -allowProvisioningUpdates due to error: " + e);
+		}
+		if (this.$logger.getLevel() === "INFO") {
+			localArgs.push("-quiet");
+			this.$logger.info(`xcodebuild ${localArgs.join(" ")}`);
+		}
+		return await this.$childProcess.spawnFromEvent("xcodebuild",
+			localArgs,
 			"exit",
-			{ stdio: buildConfig.buildOutputStdio || "inherit", cwd: this.getPlatformData(projectData).projectRoot },
+			{ stdio: stdio || "inherit", cwd },
 			{ emitOptions: { eventName: constants.BUILD_OUTPUT_EVENT_NAME }, throwError: true });
-		// this.$logger.out("xcodebuild build succeded.");
+	}
 
-		await this.createIpa(projectRoot, projectData, buildConfig);
+	private async setupSigningFromTeam(projectRoot: string, projectData: IProjectData, teamId?: string) {
+		const xcode = this.$pbxprojDomXcode.Xcode.open(this.getPbxProjPath(projectData));
+		const signing = xcode.getSigning(projectData.projectName);
+
+		let shouldUpdateXcode = false;
+		if (signing && signing.style === "Automatic") {
+			if (signing.team !== teamId) {
+				// Maybe the provided team is name such as "Telerik AD" and we need to convert it to CH******37
```

Comment:
The code here will do the following:
It will check if the arg passed to team-id matches the team in the Automatic signing style in Xcode. If it does, the native project and prepare are up-to-date.
If it does not - it will consider the team-id to be actually a team name, it will check for a team with name matching the provided team-id and if it finds one it will check if Xcode is configured to sign with that team's id.

This can fail if you have a team that has team id matching one of your other team's names, that's highly unlikely. If this raises as an issue we can split a team-id and team-name switches.